### PR TITLE
Refactorings

### PR DESCRIPTION
**Pull request by purplefox**
@vietj @alexlehm please review

### CODE REVIEW

File: vertx-mail-client/src/main/java/io/vertx/ext/mail/MailConfig.java [hunk 1]
```diff
@@ -371,21 +353,21 @@ public MailConfig setTrustAll(boolean trustAll) {
 
   /**
    * get the NetClientOptions to be used when opening SMTP connections
-   * 
+   *
    * @return the netClientOptions
    */
+  // FIXME - why allow NetClientOptions to be provided?
```
Comment (alexlehm) [inline vertx-mail-client/src/main/java/io/vertx/ext/mail/MailConfig.java]: when using a custom key store, the NetClientOptions are necessary to the set the correct jks options, see this unit test:
https://github.com/vert-x3/vertx-mail-client/blob/master/src/test/java/io/vertx/ext/mail/MailLocalTest.java#L46

File: vertx-mail-client/src/main/java/io/vertx/ext/mail/impl/MailClientImpl.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      handleError("mail service has been closed", resultHandler, context);
+    }
+    return this;
+  }
+
+  private void sendMessage(MailMessage email, SMTPConnection conn, Handler<AsyncResult<JsonObject>> resultHandler,
+                           Context context) {
+    new SMTPSendMail(conn, email, v -> {
+      conn.returnToPool();
+      /// FIXME Why return a JSON object? What's the point?
```
Comment (alexlehm) [inline vertx-mail-client/src/main/java/io/vertx/ext/mail/impl/MailClientImpl.java]: when sending mails to multiple recipients it is possible that some addresses work and some don't, for this an array or json object can be returned with the list of successful addresses and the list of failed addresses. this is not implemented yet

**Comment by alexlehm** [comment]
will redo the pr for a new wip branch from master

**Comment by purplefox** [comment]
Why did you close this Alex? I was waiting for you review...

**Pull request closed without merging**